examples/server-parallel/server.cpp

```diff
@@ -0,0 +1,939 @@
+                      res.set_header("Access-Control-Allow-Origin", "*");
+                      });
+
+    svr.Get("/", [&](const Request & /*req*/, Response &res)
+            { res.set_content(index_html_, "text/html"); });
+
+    svr.Get("/index.js", [&](const Request & /*req*/, Response &res)
+            { res.set_content(index_js_, "text/html"); });
+
+    svr.Get("/props", [&llama](const Request & /*req*/, Response &res)
+            {
+                res.set_header("Access-Control-Allow-Origin", "*");
+                json data = {
+                    { "user_name", llama.user_name.c_str() },
+                    { "assistant_name", llama.assistant_name.c_str() }
+                };
+                res.set_content(data.dump(), "application/json"); });
+
+    svr.Get("/cancel", [&llama](const Request &  req/*req*/, Response &res) {
+                res.set_header("Access-Control-Allow-Origin", "*");
+                if(req.has_param("slot_id")) {
+                    int slot_id = std::stoi(req.get_param_value("slot_id"));
+                    string result = "done";
+                    if(!llama.releaseSlot(slot_id)) {
+                        result = "wrong slot ID";
+                    }
+                    json data = {
+                        { "status", result }
+                    };
+                    res.set_content(data.dump(), "application/json");
+                } else {
+                    json data = {
+                        { "error", "Missing parameter" }
+                    };
+                    res.set_content(data.dump(), "application/json");
+                }
+                 });
+
+    svr.Post("/completion", [&llama](const Request &req, Response &res)
+             {
+                res.set_header("Access-Control-Allow-Origin", "*");
+        llama_client_slot* slot = llama.requestCompletion(json::parse(req.body));
+        // Verify if the slot exist
+        if (slot) {
+            res.set_chunked_content_provider("text/event-stream",
+                [slot](size_t /*offset*/, DataSink &sink) {
+                    if(slot->available()) { // slot has been released
+                        sink.done();
+                        return false;
+                    }
+                    if(slot->hasNewToken()) { // new token notification
+                        stringstream ss;
+                        json res_d = {
+                            { "content", slot->sampled_tokens.back() },
+                            { "slot_id", slot->id }};
+                        slot->sampled_tokens.pop_back();
+                        ss << "data: " << res_d.dump() << "\n\n";
+                        string result = ss.str();
+                        if(!sink.write(result.c_str(), result.size())) {
+                            slot->release();
+                            return false;
+                        }
+                    }
+                    return true;
+                });
```

Comment:
```suggestion
                },
                [slot](bool) {
                    slot->release();
                });
```

For abort issue, I think it missing a complete handler.

Reply:
This works!! thank you for this idea 😊